src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -380,29 +395,132 @@ internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<Functi
+                {
+                    config.Add(Connection, nameValue.ToString());
+                }
+            }
+            catch (Exception)
+            {
+                throw new InvalidDataException(InvalidDurableV1ConfigErrorMsg);
+            }
+
+            return config;
+        }
+
+        private Dictionary<string, string> GetDurableV2Config(JObject durableHostConfig)
+        {
+            var config = new Dictionary<string, string>();
+            try
+            {
+                if (durableHostConfig.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+                {
+                    config.Add(HubName, nameValue.ToString());
+                }
+                else
+                {
+                    string hubName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteName);
+                    hubName = SanitizeHubName(hubName);
+                    config.Add(HubName, hubName);
+                }
+
+                if (durableHostConfig.TryGetValue(DurableTaskV2StorageOptions, StringComparison.OrdinalIgnoreCase, out JToken storageOptions) && (storageOptions as JObject) != null)
+                {
+                    if (((JObject)storageOptions).TryGetValue(DurableTaskV2StorageConnectionName, StringComparison.OrdinalIgnoreCase, out nameValue) && nameValue != null)
                     {
-                        throw new InvalidDataException("Invalid host.json configuration for 'durableTask'.");
+                        config.Add(Connection, nameValue.ToString());
                     }
                 }
             }
+            catch (Exception)
+            {
+                throw new InvalidDataException("Invalid host.json configuration for 'durableTask' for durable task extension 2.x");
+            }
 
             return config;
         }
 
+        private bool HasDurableV1SpecificConfig(JObject durableHostConfig)
+        {
+            if (durableHostConfig.TryGetValue(DurableTaskV1StorageConnectionName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool HasDurableV2SpecificConfig(JObject durableHostConfig)
+        {
+            if (durableHostConfig.TryGetValue(DurableTaskV2StorageOptions, StringComparison.OrdinalIgnoreCase, out JToken storageOptions) && (storageOptions as JObject) != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        // Since in Durable V2 we default to sitename, it is possible that the sitename is not a valid task hub name.
+        // This algorithm translates any sitename into a valid taskhub name for Azure Storage.
+        // This logic is copied from DurableTask extension. Ideally we would put this somewhere shared.
```

Comment:
I didn't find this logic in the extension repo - I only found [this](https://github.com/Azure/azure-functions-durable-extension/blob/eb186eadb73a21d0efdc33cd7603fde5d802cab9/src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs#L42)?

An alternative fix that would avoid duplicating this code would be for you to update scale controller so it defaults correctly for v2. Currently it's already defaulting for v1.

Reply:
I have updated the comments to be more clear about where this logic lives. It unfortunately lives in two locations, as the sanitizing logic is AzureStorage specific, and we may allow for other storage providers in the future.